fe/fe-core/src/main/java/com/starrocks/persist/metablock/SRMetaBlockReaderV1.java

```diff
@@ -117,14 +111,111 @@ public long readLong() throws IOException, SRMetaBlockEOFException {
+        return readJson(String.class);
+    }
+
+    @Override
+    public <T> T readJson(Type returnType) throws IOException, SRMetaBlockEOFException {
         byte[] bytes = readJsonBytes();
-        try (JsonReader jsonReader = new JsonReader(new InputStreamReader(new ByteArrayInputStream(bytes),
-                StandardCharsets.UTF_8))) {
+        try (JsonReader jsonReader = genJsonReader(bytes)) {
             return GsonUtils.GSON.fromJson(jsonReader, returnType);
         }
     }
 
+    public <T> T readJson(Class<T> classOfT) throws IOException, SRMetaBlockEOFException {
+        Object object = readJson((Type) classOfT);
+        return Primitives.wrap(classOfT).cast(object);
+    }
+
+    @Override
+    public <T> void readCollection(Class<T> classType, CollectionConsumer<? super T> action)
+            throws IOException, SRMetaBlockEOFException {
+        int size = readInt();
+        while (size-- > 0) {
+            byte[] bytes = readJsonBytes();
+            try (JsonReader jsonReader = genJsonReader(bytes)) {
+                T t = GsonUtils.GSON.fromJson(jsonReader, classType);
+                action.accept(t);
+            } catch (SubtypeNotFoundException e) {
+                if (Config.ignore_unknown_subtype) {
+                    LOG.warn("ignore unknown sub type {}", e.getSubtype(), e);
+                } else {
+                    throw e;
+                }
+            }
+        }
+    }
+
+    private JsonReader genJsonReader(byte[] bytes) {
+        return new JsonReader(new InputStreamReader(new ByteArrayInputStream(bytes),
+                StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public <K, V> void readMapEntry(Type keyType, Type valueType,
+                                    MapEntryConsumer<? super K, ? super V> action)
+            throws IOException, SRMetaBlockEOFException {
+        int size = readInt();
+        while (size-- > 0) {
+            K k = null;
+            byte[] bytes = readJsonBytes();
+            try (JsonReader jsonReader = genJsonReader(bytes)) {
+                k = GsonUtils.GSON.fromJson(jsonReader, keyType);
+            } catch (SubtypeNotFoundException e) {
+                if (Config.ignore_unknown_subtype) {
+                    LOG.warn("ignore unknown sub type {}", e.getSubtype(), e);
+                } else {
+                    throw e;
+                }
+            }
+            V v = null;
+            bytes = readJsonBytes();
+            try (JsonReader jsonReader = genJsonReader(bytes)) {
+                v = GsonUtils.GSON.fromJson(jsonReader, valueType);
+            } catch (SubtypeNotFoundException e) {
+                if (Config.ignore_unknown_subtype) {
+                    LOG.warn("ignore unknown sub type {}", e.getSubtype(), e);
+                } else {
+                    throw e;
+                }
+            }
+            if (k != null && v != null) {
```

Comment:
is it possible that the real value is null?

Reply:
If there is a subtype that not found, and ignore_unknown_subtype is true, the value will be null.